2 A study of the strengths of Ivy League football teams shows that if a school
has a strong team one year it is equally likely to have a strong team or average
team next year; if it has an average team, half the time it is average next year,
and if it changes it is just as likely to become strong as weak; if it is weak it
has 2/3 probability of remaining so and 1/3 of becoming average.
(a) A school has a strong team. On the average, how long will it be before
it has another strong team?
(b) A school has a weak team; how long (on the average) must the alumni
wait for a strong team?

In [8]:
import numpy as np

# Transition matrix
# States: 0 = Strong, 1 = Average, 2 = Weak
P = np.array([[0.5, 0.5, 0.0],  # Strong to Strong, Average, Weak
              [0.25, 0.5, 0.25],  # Average to Strong, Average, Weak
              [0.0, 1/3, 2/3]])  # Weak to Strong, Average, Weak

# (a) Expected time to return to Strong state from Strong
# Solve (I - Q + 1)^T * 1 = 1, where Q is the submatrix of P without the Strong to Strong transition
Q_a = P[1:, 1:]

I_a = np.eye(len(Q_a))
ones_a = np.ones(len(Q_a))
expected_time_strong = np.linalg.inv(I_a - Q_a + np.ones((2, 2))).dot(ones_a)
expected_time_strong



array([0.38888889, 0.55555556])

In [9]:
# (b) Expected time to reach Strong state from Weak state
# Modify the transition matrix to consider Weak as the initial state and Strong as absorbing
P_modified = np.array([[2/3, 1/3, 0.0],  # Weak to Weak, Average, Strong
                       [0.25, 0.5, 0.25],  # Average to Weak, Average, Strong
                       [0.0, 0.0, 1.0]])  # Strong to Weak, Average, Strong (absorbing state)
Q_b = P_modified[:2, :2]
I_b = np.eye(len(Q_b))
expected_time_weak_to_strong = np.linalg.inv(I_b - Q_b).dot(ones_a[:2])

expected_time_weak_to_strong

array([10.,  7.])

For a school with a strong team, on average, it will be approximately 0.39 years before it has another strong team
 For a school with a weak team, alumni must wait on average 10 years for the team to become strong again.